recipes/mimalloc/all/conanfile.py

```diff
@@ -0,0 +1,149 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class MimallocConan(ConanFile):
+    name = "mimalloc"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/microsoft/mimalloc"
+    description = "mimalloc is a compact general purpose allocator with excellent performance."
+    topics = ("conan", "mimalloc", "allocator", "performance", "microsoft")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "secure": [True, False],
+        "override": [True, False],
+        "single_object": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "secure": False,
+        "override": False,
+        "single_object": False
+    }
+    generators = "cmake"
+    exports_sources = "CMakeLists.txt", "patches/*"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("mimalloc-" + self.version, self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        tools.check_min_cppstd(self, "14")
+        if self.options.shared:
+            del self.options.fPIC
+
+        if self.options.shared or self.settings.compiler == "Visual Studio":
+            del self.options.single_object
+
+        if self.options.override:
+            if self.options.shared and self.settings.compiler == "Visual Studio" and \
+                    "MT" in str(self.settings.compiler.runtime):
+                raise ConanInvalidConfiguration(
+                    "Cannot use MT(d) runtime when building mimalloc as a shared library for override")
+
+    _cmake = None
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        if self._cmake.is_multi_configuration:
+            self._cmake.definitions["CMAKE_BUILD_TYPE"] = self.settings.build_type
+        self._cmake.definitions["MI_BUILD_TESTS"] = False
+        self._cmake.definitions["MI_BUILD_SHARED"] = self.options.shared
+        self._cmake.definitions["MI_BUILD_STATIC"] = not self.options.shared and not self.options.get_safe("single_object", False)
+        self._cmake.definitions["MI_BUILD_OBJECT"] = self.options.get_safe("single_object", False)
```

Comment:
Does this mean that `shared=True and single_object=True` is valid?

Reply:
Actually no.

Reply:
Fixed in 1b8ed793